### PR TITLE
SW-650_allow_custom_criteria

### PR DESCRIPTION
**Pull request by TheKeymaster**

### CODE REVIEW

File: tests/Core/Content/Product/SalesChannel/Listing/ProductRouteBase.php [hunk 1]
```diff
@@ -197,17 +199,31 @@ public function testShopwareHandlesRequestWhenFindologicisInactive(): void
...
+    {
+        $salesChannelContextMock = $this->getMockedSalesChannelContext(true);
+        $request = Request::create('http://your-shop.de/some-category');
+        $request->setSession($this->getSessionMock());
+
+        $criteria = new Criteria();
+        $criteria->addFilter(new EqualsFilter('product.name', 'yeet'));
+
+        $productRoute = $this->getRoute();
+        $this->call($productRoute, $request, $salesChannelContextMock, '69', $criteria);
```
Comment (dblaichinger) [inline tests/Core/Content/Product/SalesChannel/Listing/ProductRouteBase.php:212]: This test seem to not assert anything or check any method calls, can this be improved?

**Review by dblaichinger — CHANGES_REQUESTED** [review]

File: tests/Core/Content/Product/SalesChannel/Listing/ProductRouteBase.php (hunk 1 above)
Comment (TheKeymaster) [reply tests/Core/Content/Product/SalesChannel/Listing/ProductRouteBase.php:212]: Well not really, as before this would've resulted in an exception.

Comment (dblaichinger) [reply tests/Core/Content/Product/SalesChannel/Listing/ProductRouteBase.php:212]: So there is no way to verify `yeet` is used?

Comment (TheKeymaster) [reply tests/Core/Content/Product/SalesChannel/Listing/ProductRouteBase.php]: Sure, I used the response to ensure the criteria contains the filter.

**Review by dblaichinger — APPROVED** [review]

**Pull request merged**